### PR TITLE
Cleanup test certificate temp files

### PR DESCRIPTION
**Pull request by fvaleri**
Some tests leave a bunch of temporary files in the /tmp folder, which are used for certificate generation. This change adds code to clean them up after the test has finished. It also hides ClientAndServer in MockCruiseControl to isolate test classes from the mockserver dependency.

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
Why not just mark the files as `deleteOnExit()`?

**Comment by fvaleri** [comment]
> `deleteOnExit`

You are right, that's better. Thanks.

I would like to keep the ClientAndServer wrapping in MockCruiseControl, which is useful to isolates test classes from the mockserver dependency. Wdyt?

**Review by scholzj — APPROVED** [review]
I'm fine with the MockCruiseControl change. But maybe it is worth mentioning in the PR description? Also, I wonder if one of the tests has some more places to check. But LGTM otherwise. Thanks.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/CaReconcilerTest.java
[the remark this answers is not in the document]
Comment (fvaleri) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/CaReconcilerTest.java:1]: Right, and there were also sole File.delete calls. I'm always using deleteOnExit now.

**Review by ppatierno — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**